### PR TITLE
fix: move user export to direct download

### PR DESCRIPTION
**Pull request by ludtkemorgan**
# Pull Request Template

## Issue Overview

This PR addresses #issue

- [ ] This change addresses the issue in full
- [x] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

This PR tackles the following issues found during the most recent bug bash:
- User export having incorrect email text. This was solved by moving it from an email export to a direct download by using the same patterns as applications and listings exports.
- If you update your birthday as a public user and leave a field empty it was auto-populating it with a value (e.g 2001 for year and 01 for both month and day). This has been solved by making the field required on the frontend so the proper ui errors appear.
- Going to a submitted application from the user profile page was sometimes throwing errors. This was fixed with a null check on the FormSummaryDetails page.

~Note: This does not address the issue with some applications not getting tied to user accounts. That is still being addressed and will be in a follow-up PR~
I have been unable to reproduce this

## How Can This Be Tested/Reviewed?

The following flows should be tested:
1. Logging in as admin and exporting user csv
2. Logging in as a jurisdictional admin and exporting user csv
3. Create a public profile and try to edit the birthday in the settings tab. Deleting any of the fields and attempting to save should throw an error

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue with applicable URLs
- [ ] I have performed a self-review of my own code
- [ ] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

**Review by YazeedLoonat — APPROVED** [review]
LGTM

File: shared-helpers/src/types/backend-swagger.ts [hunk 1]
```diff
@@ -1580,7 +1580,7 @@ export class UserService {
   /**
    * List users in CSV
    */
-  listAsCsv(options: IRequestOptions = {}): Promise<SuccessDTO> {
+  listAsCsv(options: IRequestOptions = {}): Promise<any> {
```
Comment (ColinBuyck) [inline shared-helpers/src/types/backend-swagger.ts:1583]: Is there a reason this wasn't generated to be StreamableFile?

**Review by ColinBuyck — APPROVED** [review]
LGTM 🥅 Just one question for my own understanding

File: shared-helpers/src/types/backend-swagger.ts (hunk 1 above)
Comment (ludtkemorgan) [reply shared-helpers/src/types/backend-swagger.ts:1583]: Good question. This typing is picked up from the `@ApiOkResponse()` field which we aren't adding to the `listAsCsv` endpoints. I decided to keep this in sync with the other csv endpoints. But that is something we can look to update in the future

**Pull request merged**